### PR TITLE
 HMAN-394: Update processing of inbound adaptor when the hearing resp…

### PR DESCRIPTION
**Pull request by Thor-tech-of-metal**
HMAN-394: Update processing of inbound adaptor when the hearing response from LA is "awaiting listing"
            [HMAN-394] (https://tools.hmcts.net/jira/browse/HMAN-394).

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceImpl.java [hunk 1]
```diff
@@ -44,11 +45,25 @@ public HearingManagementServiceImpl(final MessageSenderConfiguration messageSend
 
     @Override
     public void processRequest(String caseId, HearingDetailsRequest hearingDetailsRequest) {
-        Integer latestVersion = cftHearingService.getLatestVersion(caseId);
+
+        val latestVersion = cftHearingService.getLatestVersion(caseId);
+        if (isAwaitingStatus(hearingDetailsRequest)) {
+            log.warn("Hearing response received for hearing ID {} with hearingCaseStatus 5 (Awaiting Listing)",
+                     latestVersion);
```
Comment (Thor-tech-of-metal) [inline src/main/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceImpl.java:52]: info

File: src/main/java/uk/gov/hmcts/reform/hmc/client/model/hmi/HearingCode.java [hunk 2]
```diff
@@ -5,7 +5,8 @@
 public enum HearingCode {
     LISTED(100, "LISTED"),
     PENDING_RELISTING(6, "PENDING_RELISTING"),
-    CLOSED(8, "CLOSED")
+    CLOSED(8, "CLOSED"),
+    AWAITING_TO_BE_LISTED(5, "AWAITING_TO_BE_LISTED")
```
[the remark this answers is not in the document]
Comment (Thor-tech-of-metal) [reply src/main/java/uk/gov/hmcts/reform/hmc/client/model/hmi/HearingCode.java:9]: SI SI DONE

File: src/main/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceImpl.java [hunk 3]
```diff
@@ -44,11 +45,25 @@ public HearingManagementServiceImpl(final MessageSenderConfiguration messageSend
 
     @Override
     public void processRequest(String caseId, HearingDetailsRequest hearingDetailsRequest) {
-        Integer latestVersion = cftHearingService.getLatestVersion(caseId);
+
+        val latestVersion = cftHearingService.getLatestVersion(caseId);
+        if (isAwaitingStatus(hearingDetailsRequest)) {
```
[the remark this answers is not in the document]
Comment (Thor-tech-of-metal) [reply src/main/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceImpl.java:50]: DONE

File: src/main/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Thor-tech-of-metal) [reply src/main/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceImpl.java:52]: DONE

Comment (Thor-tech-of-metal) [reply src/main/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceImpl.java:52]: DONE

File: src/main/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceImpl.java [hunk 4]
```diff
@@ -44,9 +45,24 @@ public HearingManagementServiceImpl(final MessageSenderConfiguration messageSend
 
     @Override
     public void processRequest(String caseId, HearingDetailsRequest hearingDetailsRequest) {
-        Integer latestVersion = cftHearingService.getLatestVersion(caseId);
+
+        val hearingId = cftHearingService.getLatestVersion(caseId);
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceImpl.java:49]:
```suggestion
        val latestVersion = cftHearingService.getLatestVersion(caseId);
```
Sorry, this one is still `latestVersion`, but `caseId` is also known as the `hearingId` 😄

**Review by danlysiak — APPROVED** [review]

**Pull request merged**